addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js

```diff
@@ -0,0 +1,195 @@
+        this.input = useRef("section_and_note_input");
+    }
+
+    async loadOptionsSource(request) {
+        const options = await super.loadOptionsSource(request);
+        if (request.length) {
+            options.unshift({
+                label: _t('Label "%s"', request),
+                classList: "o_m2o_dropdown_option",
+                action: () => {
+                    this.props.updateLabel(request);
+                },
+            });
+        }
+        return options;
+    }
+
+    onKeyDown(ev) {
+        if (ev.key === "Enter" && this.isSectionOrNote) {
+            this.props.updateLabel(this.input.el.value);
+        }
+    }
+
+    get isSectionOrNote() {
+        return this.props.isSection || this.props.isNote;
+    }
+}
+
+export class ProductLabelSectionAndNoteField extends Many2OneField {
+    static components = {
+        ...Many2OneField.components,
+        Many2XAutocomplete: ProductLabelSectionAndNoteFieldAutocomplete,
+    };
+    static template = "account.ProductLabelSectionAndNoteField";
+
+    setup() {
+        super.setup();
+        this.extended = useState({ value: false });
+        this.inputRef = useRef("autoResizeInput");
+        useAutoresize(this.inputRef);
+    }
+
+    get displayName() {
+        const { product_id, name } = this.props.record.data;
+        if (!product_id && !name) {
+            return "";
+        } else if (!product_id && name) {
+            return typeof name === "string" ? name : super.displayName;
+        } else {
+            return super.displayName;
+        }
+    }
+
+    get extraLines() {
+        const { product_id, name } = this.props.record.data;
+        if (name && ((product_id && this.extended.value) || !product_id)) {
+            return typeof name === "string" ? name : this.value[1] ? this.value[1] : [];
+        }
+        return false;
+    }
+
+    get hasProduct() {
+        return Boolean(this.props.record.data?.product_id);
+    }
+
+    get Many2XAutocompleteProps() {
+        const props = super.Many2XAutocompleteProps;
+        props.updateLabel = this.updateLabel.bind(this);
+        props.placeholder = _t("Chose a product/Write a label");
+        props.isSection = this.isSection(this.props.record);
+        props.isNote = this.isNote(this.props.record);
+        return props;
+    }
+
+    get isProductClickable() {
+        return this.props.record.model.root.data.state !== "draft";
+    }
+
+    toggleExtended() {
+        this.extended.value = !this.extended.value;
+    }
+
+    updateLabel(value) {
+        this.props.record.update({ name: value });
+        this.extended.value = true;
+    }
+
+    isSection(record = null) {
+        record = record || this.props.record;
+        return record.data.display_type === "line_section";
+    }
+
+    isNote(record = null) {
+        record = record || this.props.record;
+        return record.data.display_type === "line_note";
+    }
+
+    getSectionAndNoteClasses() {
```

Comment:
I know it is kinda sus but i felt like it was either that or a huuuuge t-if t-else so i figured a dynamic class was better since it would be the only difference between the elements from the t-if and the ones from the t-else